cardano-api/test/Test/Cardano/Api/TxBody.hs

```diff
@@ -0,0 +1,285 @@
+  SimpleScript SimpleScriptV2 $ upgradeSimpleScript script
+
+upgradeSimpleScript ::
+  SimpleScript SimpleScriptV1 -> SimpleScript SimpleScriptV2
+upgradeSimpleScript = \case
+  RequireSignature hash -> RequireSignature hash
+  RequireAllOf scripts  -> RequireAllOf $ map upgradeSimpleScript scripts
+  RequireAnyOf scripts  -> RequireAnyOf $ map upgradeSimpleScript scripts
+  RequireMOf n scripts  -> RequireMOf n $ map upgradeSimpleScript scripts
+
+
+-- * View: Change 'TxBodyContent' “phase” to 'ViewTx'
+--
+-- Basically, it's just dropping witness information.
+
+viewBodyContent :: TxBodyContent BuildTx era -> TxBodyContent ViewTx era
+viewBodyContent body =
+  TxBodyContent
+    { txAuxScripts      =                    txAuxScripts     body
+    , txCertificates    = viewCertificates $ txCertificates   body
+    , txExtraKeyWits    =                    txExtraKeyWits   body
+    , txExtraScriptData = ViewTx
+    , txFee             =                    txFee            body
+    , txIns             = map viewTxIn     $ txIns            body
+    , txInsCollateral   =                    txInsCollateral  body
+    , txMetadata        =                    txMetadata       body
+    , txMintValue       = viewMintValue    $ txMintValue      body
+    , txOuts            =                    txOuts           body
+    , txProtocolParams  = ViewTx
+    , txUpdateProposal  =                    txUpdateProposal body
+    , txValidityRange   =                    txValidityRange  body
+    , txWithdrawals     = viewWithdrawals  $ txWithdrawals    body
+    }
+
+viewTxIn ::
+  (TxIn, BuildTxWith BuildTx (Witness WitCtxTxIn era)) ->
+  (TxIn, BuildTxWith ViewTx  (Witness WitCtxTxIn era))
+viewTxIn = second $ const ViewTx
+
+viewWithdrawals :: TxWithdrawals BuildTx era -> TxWithdrawals ViewTx era
+viewWithdrawals = \case
+  TxWithdrawalsNone                 -> TxWithdrawalsNone
+  TxWithdrawals support withdrawals ->
+    TxWithdrawals
+      support
+      [(address, amount, ViewTx) | (address, amount, _) <- withdrawals]
+
+viewCertificates :: TxCertificates BuildTx era -> TxCertificates ViewTx era
+viewCertificates = \case
+  TxCertificatesNone                    -> TxCertificatesNone
+  TxCertificates support certificates _ ->
+    TxCertificates support certificates ViewTx
+
+viewMintValue :: TxMintValue BuildTx era -> TxMintValue ViewTx era
+viewMintValue = \case
+  TxMintNone                  -> TxMintNone
+  TxMintValue support value _ -> TxMintValue support value ViewTx
+
+
+-- * Build: Change 'TxBodyContent' “phase” to 'BuildTx'
+--
+-- Here we make up the most trivial witnesses or plug holes with 'panic'
+-- to make sure the fields are not touched.
+
+buildBodyContent :: TxBodyContent ViewTx era -> TxBodyContent BuildTx era
+buildBodyContent body =
+  TxBodyContent
+    { txAuxScripts      =                     txAuxScripts     body
+    , txCertificates    = buildCertificates $ txCertificates   body
```

Comment:
Alignment

Reply:
Do you think misalignment will make it more maintainable?

Reply:
done